### PR TITLE
Updates to Shape Selection tool

### PR DESCRIPTION
**Pull request by david-fry**
Updated the shape selection tool to allow for editing text after selecting the text shape. Also added resizing of certain shapes. 

Added Update Shape and Delete Shape action. Undo still needs to be completed for these items.

### CODE REVIEW

File: src/tools/SelectShape.coffee [hunk 1]
```diff
@@ -12,64 +29,337 @@ module.exports = class SelectShape extends Tool
...
+    switch @currentShapeState
+      when 'selected'        
+        lc.setShapesInProgress([@_getSelectionShape(lc.ctx), @selectedShape])      
+      when 'editing'
+        lc.setShapesInProgress([@_getSelectionShape(lc.ctx, '#fff')])       
+      else
+        lc.setShapesInProgress([@selectedShape])
+
+
+  _updateInputEl: (lc, withMargin=false) ->
```
Comment (irskep) [inline src/tools/SelectShape.coffee:60]: I'm not comfortable with this level of code duplication. This is exactly what appears in the text tool, right? There needs to be some kind of input box management module or class.

**Comment by irskep** [comment]
Very cool! I do have some concerns about duplicate code, so I'm not going to merge this yet.

**Comment by stto3703** [comment]
@david-fry is it possible to get the resize behavior for line on pencil?

**Comment by irskep** [comment]
FYI, for anyone reading this in The Future, I never merged this because it has a bunch of copypasta from the text tool, which is not good.

**Pull request closed without merging**